homeassistant/components/ps4/__init__.py

```diff
@@ -37,21 +39,56 @@
     data = entry.data
     version = entry.version
 
-    reason = {1: "Region codes have changed"}  # From 0.89
+    _LOGGER.debug("Migrating PS4 entry from Version %s", version)
 
-    # Migrate Version 1 -> Version 2
+    reason = {
+        1: "Region codes have changed",
+        2: "Format for Unique ID for entity registry has changed"
+    }
+
+    # Migrate Version 1 -> Version 2: New region codes.
     if version == 1:
         loc = await hass.async_add_executor_job(location.detect_location_info)
         if loc:
             country = loc.country_name
             if country in COUNTRIES:
                 for device in data['devices']:
                     device[CONF_REGION] = country
-                entry.version = 2
+                version = entry.version = 2
                 config_entries.async_update_entry(entry, data=data)
                 _LOGGER.info(
                     "PlayStation 4 Config Updated: \
                     Region changed to: %s", country)
+
+    # Migrate Version 2 -> Version 3: Update identifier format.
+    if version == 2:
+        # Prevent changing entity_id. Updates entity registry.
+        registry = await entity_registry.async_get_registry(hass)
+
+        for entity_id, e_entry in registry.entities.items():
+            if e_entry.config_entry_id == entry.entry_id:
+                unique_id = e_entry.unique_id
+
+                # Remove old entity entry.
+                registry.async_remove(entity_id)
+                await hass.async_block_till_done()
```

Comment:
This line blocked the HA startup I believe

Reply:
That means, this PR was never tested....

Reply:
🙈

That's bad. Wonder how I missed that.

Reply:
Ah yeah, I reported that one but never filed an issue for it, so I also get some blame :(

Reply:
#23893

Reply:
Oops. My bad.